sys/mips/timer.c

```diff
@@ -110,8 +111,7 @@ static int mips_timer_probe(device_t *dev) {
 static int mips_timer_attach(device_t *dev) {
   mips_timer_state_t *state = dev->state;
 
-  state->intr_handler =
-    INTR_HANDLER_INIT(mips_timer_intr, NULL, dev, "MIPS CPU timer", 0);
+  state->irq_res = bus_alloc_irq(dev, 0, MIPS_HWINT5, RF_ACTIVE);
```

Comment:
`assert(state->irq_res)`

Reply:
Let's not be overly defensive. `NULL` dereference crashes kernel immediately and is easy to debug.